datastore/postgres/updatevulnerabilities.go

```diff
@@ -51,9 +51,48 @@ var (
 // provided vulnerabilities and computes a diff comprising the removed
 // and added vulnerabilities for this UpdateOperation.
 func (s *MatcherStore) UpdateVulnerabilities(ctx context.Context, updater string, fingerprint driver.Fingerprint, vulns []*claircore.Vulnerability) (uuid.UUID, error) {
+	return s.updateVulnerabilities(ctx, updater, fingerprint, vulns, nil, false)
+}
+
+// DeltaUpdateVulnerabilities implements vulnstore.Updater.
+//
+// It shares functionality with UpdateVulnerabilities but with delta
+// support, so the process goes:
+// * Create a new UpdateOperation
+// * Query existing vulnerabilities for the updater
+// * Discount and vulnerabilities with newer updates and deleted vulnerabilities
+// * Update the associated updateOperation for the remaining existing vulnerabilities
+// * Insert the new vulnerabilities
+// * Associate new vulnerabilities with new updateOperation
```

Comment:
```suggestion
//   - Create a new UpdateOperation
//   - Query existing vulnerabilities for the updater
//   - Discount and vulnerabilities with newer updates and deleted vulnerabilities
//   - Update the associated updateOperation for the remaining existing vulnerabilities
//   - Insert the new vulnerabilities
//   - Associate new vulnerabilities with new updateOperation
```
See the "Lists" section in https://go.dev/doc/comment